### ISSUE TITLE
Expose native index client for direct queries

### ISSUE THREAD

**Issue by wojciechwojcik** [JanusGraph/janusgraph#338]
See #318. Still there are more use cases that require direct read-access to index. 

E.g. with Solr if you want to use different parser than standard lucene query parser e.g. dismax or edismax, there are many more fields that can be set that just the query field:
https://cwiki.apache.org/confluence/display/solr/The+DisMax+Query+Parser
https://cwiki.apache.org/confluence/display/solr/The+Extended+DisMax+Query+Parser

With current ``janusGraph.indexQuery(String queryText)`` approach utilizing such features is impossible.

**Comment by wojciechwojcik** [comment]
Some propositions:

A) get native client directly for read-only operations
```
interface Trasaction {
   /** existing interface: just indexName and queryString, all results are always fetched */
    public JanusGraphIndexQuery indexQuery(String indexName, String query);

   /** proposed interface: */
   public JanusGraphNativeIndexQueryHelper indexNativeQuery(String indexBackendName)
}

class JanusGraphNativeIndexQueryHelper {

    /**
    @returns depending on backend type an instance of:
      org.apache.solr.client.solrj.SolrClient
      org.elasticsearch.client.Client

      The client should only be used for read-only query operations.
      This could be used to facilitate count only queries, or setting extra request fields other than just the query string.
    @throws UnsupportedOperationException for embeded lucene backend
    @throws ClassCastException in case of type mismatch
    */
    public <T> T getNativeIndexClient(String indexBackendName)  
  
   /** translate janus management index name to index store name */
   public String getNativeIndexName(String indexName)

   /** substitute v.x, p.x, e.x occurrences in the query, same way org.janusgraph.graphdb.database.IndexSerializer::execute() does */
   public String transtaleQuery(String query)
}
```

B) fix existing extra parameter passing framework, which currently does nothing + allow count only queries

```
org.janusgraph.graphdb.query.graph.IndexQueryBuilder::addParameter()
org.janusgraph.graphdb.query.graph.IndexQueryBuilder::getParameters()
org.janusgraph.diskstorage.indexing.RawQuery::getParameters()
```

This should be used e.g in
``org.janusgraph.diskstorage.solr.SolrIndex::query(RawQuery query)``
to pass further down to: ``SolrQuery::setParam()``

Example parameters (single value, may be comma separated):
````
"defType": "edismax",
 "fl": "*,score"
"qf": "main, extra"
````

This should also support parameters with distinct multiple values e.g.:
````
      "defType": [
        "dismax",
        "edismax"
      ]
````

Count only interface:
````
interface JanusGraphIndexQuery {
/** existing methods that fetch all results: */
vertices();
edges();
properties();

/** proposed: add option to count only the results in an optimized fashion, without fetching them one by one */
count();
}
````

**Comment by wojciechwojcik** [comment]
@holleyism see proposed changes above

**Comment by wojciechwojcik** [comment]
Resolved by #423
